prime-angular-frontend/src/app/modules/adjudication/shared/guards/adjudication.guard.ts

```diff
@@ -33,35 +33,40 @@ export class AdjudicationGuard extends BaseGuard {
    * Check the user is authenticated, otherwise redirect
    * them to an appropriate destination.
    */
-  // TODO update to be two observables merged and resolved using combineLatest,
-  // but requires wrapping the Keycloak service so it uses obseravables first
   protected checkAccess(routePath: string = null): Observable<boolean> | Promise<boolean> {
-    const admin$ = this.authService.getAdmin$()
+
+    return this.authService.getAdmin$()
       .pipe(
         exhaustMap((admin: Admin) => {
-          // Attempt to create an admin if they don't exist
-          return (this.permissionService.hasRoles(Role.ADMIN))
-            ? this.adjudicationResource.createAdmin(admin)
-            : Promise.resolve(admin);
+          return this.adjudicationResource.getAdjudicatorByUserId(admin.userId)
+            .pipe(
+              exhaustMap((primeAdmin: Admin) => {
+                return of([primeAdmin, admin]);
+              }));
+        }),
+        exhaustMap(([primeAdmin, admin]: [Admin, Admin]) => {
+          if (primeAdmin) {
+            //if PRIME admin record found, pass the status to next step
+            admin.status = primeAdmin.status
+          } else {
+            // if PRIME does not have any admin user with the user id, create it
+            this.adjudicationResource.createAdmin(admin).subscribe();
+          }
```

Comment:
It seems like need to check for `this.permissionService.hasRoles(Role.ADMIN)` before creating Admin record

Reply:
From my testing, the user without admin role will not be created without the check. My testing shows that authorization-redirect.guard.ts (line 47 will check for the role) will redirect the user to denied page before hitting the code here. The check seems to be not necessary.

Reply:
You are right, this seems to be the case